### PR TITLE
Add BUILDSYSTEM_TESTING flag to enable ign-cmake

### PR DESCRIPTION
**Pull request by mjcarroll**
In support of ignitionrobotics/ign-cmake#97

Signed-off-by: Michael Carroll <michael@openrobotics.org>

### CODE REVIEW

**Comment by scpeters** [comment]
testing https://github.com/ignitionrobotics/ign-cmake/pull/97 with this branch: [Build Status](https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-windows7-amd64/317/) https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-windows7-amd64/317/

**Comment by scpeters** [comment]
I tried testing with ign-physics in order to exercise the colcon / windows workflow using https://github.com/ignition-tooling/gazebodistro/commit/4e9d64792d2af8bd7cdf1321851d7c09aa2363db, but it fails because https://github.com/ignitionrobotics/ign-cmake/pull/97 is targeting the ign-cmake master branch, which is major version 3. I'll make a backport branch of that PR to test the colcon / windows workflow

*  [Build Status](https://build.osrfoundation.org/view/ign-citadel/job/ign_physics-ign-2-win/31/) https://build.osrfoundation.org/view/ign-citadel/job/ign_physics-ign-2-win/31/

File: jenkins-scripts/lib/generic-default-devel-windows.bat [hunk 1]
```diff
@@ -82,7 +82,7 @@ if exist .\configure.bat (
 )
 
 :: We want to make sure that tests are enabled for this project's build
-cmake .. -DBUILD_TESTING:BOOL=%ENABLE_TESTS% || echo "Second run of cmake (for enabling the tests) failed"
+cmake .. -DBUILD_TESTING:BOOL=%ENABLE_TESTS% -DBUILDSYSTEM_TESTING:BOOL=%ENABLE_TESTS% || echo "Second run of cmake (for enabling the tests) failed"
```
Comment (j-rivero) [inline jenkins-scripts/lib/generic-default-devel-windows.bat:85]: I made a change to make Windows similar to Linux about injecting in a variable the parameters needed. `add_buildsystem_testing_jrivero` branch.

I've tested it here https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-windows7-amd64/318/ and here https://build.osrfoundation.org/job/ignition_cmake-ci-master-windows7-amd64/7/.

Comment (scpeters) [reply jenkins-scripts/lib/generic-default-devel-windows.bat:85]: testing your branch here: [Build Status](https://build.osrfoundation.org/view/ign-citadel/job/ign_physics-ign-2-win/33/) https://build.osrfoundation.org/view/ign-citadel/job/ign_physics-ign-2-win/33/

Comment (mjcarroll) [reply jenkins-scripts/lib/generic-default-devel-windows.bat]: I fast forwarded to include it here.

**Pull request merged**